src/main/kotlin/dev/stalla/PodcastRssParser.kt

```diff
@@ -56,17 +61,27 @@ public object PodcastRssParser {
      * and return a [Podcast] if the XML document is an RSS feed.
      *
      * @param uri The location of the content to be parsed.
-     * @return A [Podcast] if the XML document behind the URI is an RSS document, otherwise null.
+     * @return A [Podcast] if the XML document behind the URI is an RSS document, otherwise `null`.
+     * @throws IOException If any IO errors occur.
+     * @throws SAXException If any XML parsing errors occur.
+     * @throws NullPointerException If [uri] is `null`.
```

Comment:
Yes, it gets annotated and IJ shows the warning. 

In terms of contract, the only applicable one I see here is:

```
contract {
    returns() implies (uri != null)
}
```

which does not serve a real purpose since the type of `uri` is `String` (and not `String?`) anyway. In Kotlin it's trivial, and Java ignores the contract at all.

Reply:
But you gave me another idea. I totally forgot about the `callsInPlace` effect. Maybe we have a place somewhere where we can leverage something from it. Will check in the future.